integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -387,6 +389,71 @@ fn next_block_has_new_shard_layout(epoch_manager: Arc<dyn EpochManagerAdapter>,
         && shard_layout != next_shard_layout
 }
 
+/// Asserts that for each child shard (Mem)Trie and FlatState contains the same final state.
```

Comment:
@staffik Would it make sense to do the same for the State column? Or perhaps just iterate state starting from the child state root to check that all nodes are there?

Reply:
Maybe we could use `get_view_trie_for_shard` instead of `get_trie_for_shard` and the check would be almost seemless?

Reply:
Yes, that should perhaps work, view trie doesn't include memtrie for whatever reason.

Reply:
It works indeed. Expanded check to validate "disk" tries as well.